frappe/core/doctype/version/version.py

```diff
@@ -80,32 +80,38 @@ def get_diff(old, new, for_child=False):
 		updater_reference=updater_reference,
 	)
 
+	if not for_child:
+		amended_from = new.get("amended_from")
+		old_row_name_field = "_amended_from" if (amended_from and amended_from == old.name) else "name"
```

Comment:
This could be concerning (unbound variable) but only usage so far is in code block for child tables so eh.

Reply:
I thought about it, but decided to prioritise avoiding unnecessary computation for child tables.